src/Compilers/Core/Portable/SourceGeneration/Nodes/NodeStateTable.cs

```diff
@@ -182,12 +179,25 @@ public bool TryUseCachedEntries()
                 }
 
                 var previousEntries = _previous._states[_states.Count];
-                Debug.Assert(previousEntries.All(e => e.state == EntryState.Cached));
+                Debug.Assert(previousEntries.IsCached);
 
                 _states.Add(previousEntries);
                 return true;
             }
 
+            public bool TryUseCachedEntries(out ImmutableArray<T> usedEntries)
+            {
+                if (TryUseCachedEntries())
+                {
+                    usedEntries = _states[_states.Count - 1].ToImmutableArray();
+                    return true;
+                }
+                {
```

Comment:
>{ [](http://example.com/codeflow?start=16&length=1)

Consider adding `else` or removing the `{ }`.

Reply:
One of my weirder typos i'll be honest :)

---
In reply to: [638368059](https://github.com/dotnet/roslyn/pull/53568#discussion_r638368059) [](ancestors = 638368059)